Doc/library/pathlib.rst

```diff
@@ -1123,46 +1128,37 @@ call fails (for example because the path doesn't exist).
 
 .. method:: Path.is_symlink()
 
-   Return ``True`` if the path points to a symbolic link, ``False`` otherwise.
-
-   ``False`` is also returned if the path doesn't exist; other errors (such
-   as permission errors) are propagated.
+   Return ``True`` if the path points to an accessible symbolic link,
+   ``False`` otherwise. Use :meth:`Path.exists` to test if a path is
+   accessible without checking its type.
 
 
 .. method:: Path.is_socket()
 
-   Return ``True`` if the path points to a Unix socket (or a symbolic link
-   pointing to a Unix socket), ``False`` if it points to another kind of file.
-
-   ``False`` is also returned if the path doesn't exist or is a broken symlink;
-   other errors (such as permission errors) are propagated.
+   Return ``True`` if the path points to an accessible Unix socket, ``False``
```

Comment:
Okay, I'm starting to regret the "accessible" at this point 😆 But I think I've figured out why:

"Accessible" is really a property of the _path_, but these descriptions make it sound like it's a property of the _file_ (or in this case, the socket). And "inaccessible" can be ambiguous for some of these (imagine a removable drive, where the _path_ may be accessible enough to determine the type, but the _drive_ is not accessible if nothing is in there).

So new proposal (for all of them):
> Return True if the path points to a Unix socket. False may be returned if the path is invalid, inaccessible or missing, or if it points to something other than a Unix socket. Use Path.exists to distinguish between these cases.

Reply:
Isn't that a bit long? How about:

> Return True if the path is accessible and points to a valid Unix socket...

Reply:
I'm really trying to get the reference to using Path.exists in there, to really clearly lead people towards that if they care about the different results (and thereby imply that "is_*" are for when you _don't_ care about the different results).

So while I do like the shorter phrasing, I think you then end up with all the same words describing the alternative ("Use Path.exists to distinguish between cases where the path is invalid, inaccessible, missing, or points to something other ..."). Given the choice, I'd rather more clearly explain _this_ function than the alternative (or not clearly explaining either of them).

Finding a balance between reference, specification, and teaching is the toughest part of documentation, and these docs have to do all three.

Reply:
And this? If the path is accesible, it's also not missing.

> Return True if the path is accessible and points to a valid Unix socket. Use Path.exists to not check for the latter.

Reply:
It's not just not checking the type, it's also going to raise different errors based on why it's inaccessible. But I didn't want to go as far as suggesting to handle `PermissionError` vs `FileNotFoundError` vs whatever other OSErrors may be raised. 

The main thing to imply is that a `True` result from these functions implies one case, while a `False` result implies one of many cases, to prevent people thinking that `not is_file()` is the same as `is_dir()` (which is obvious to us, who are neck deep in this stuff, but is not otherwise obvious at first sight). Suggesting that `exists` will let you distinguish exactly what `False` means is the best way I've come up with so far to communicate it without making it too hard to understand.

Also, I think "Use Path.exists to not check for the latter" is very difficult to parse. But even reworded for clarity, I don't think it would communicate what I'm wanting to communicate here.

Reply:
You mean for Windows? Because for Unix the only difference is checking the file type:

https://github.com/python/cpython/blob/a7f83721d97f7b2b6c0efb21929b359c4143ed22/Lib/pathlib/_abc.py#L431-L435

https://github.com/python/cpython/blob/a7f83721d97f7b2b6c0efb21929b359c4143ed22/Lib/pathlib/_abc.py#L525-L528

Reply:
Oh, does `exists` suppress all exceptions as well? If that's the case, we should be suggesting calling `stat` instead @barneygale

Reply:
> Return True if the path points to a Unix socket. False is returned if the path is invalid, inaccessible, missing or points to something else. Use Path.stat to distinguish between these cases.